datalad_redcap/export_form.py

```diff
@@ -6,16 +6,10 @@
 from typing import (
     List,
     Optional,
-    Tuple,
 )
 
 from redcap.methods.records import Records
 
-from datalad.distribution.dataset import (
-    Dataset,
-    require_dataset,
-    resolve_path,
-)
 from datalad.interface.common_opts import (
     nosave_opt,
     save_message_opt,
```

Comment:
No related to this PR, but I want to challenge the need for this feature.

Reply:
The need for importing the parameters from `common_opts`? Or the need for having `--message` and `--nosave`?

My idea for `--nosave` was that currently the form (and also report) export can write one or multiple forms into one file. `--nosave` was supposed to allow repeating the command to write several forms into several files, and capturing them in a single, manual, save afterwards. But we could later add a many-forms-to-separate-files variant of the export command, and make the problem go away. Or we could do away with `--nosave` right now (should `--message` stay?) - is that what you suggest?

Reply:
Without deeply thinking about it again, I'd say this command should either

- always save
- never save

If it always saves, it should support setting a message (see https://github.com/datalad/datalad/issues/3316)

However, this creates complications. See https://github.com/datalad/datalad/issues/3896 for an entrypoint to some. Few more keywords are: recursive saving of superdatasets, etc.

Just saving also does not necessarily provide an indication of the origin of the save information.

Without wanting to propse a full concept: It may just be better to never save for this command.

- if people want to export multiple, they run multiple command, and save ones at the end -- no need for more code
- if people want to do things recursively, the combine this command with `for_each_dataset`, and safe at then end or individually as they desired
- if people want to have provenance capture of where that information is coming from, they combine any of the above with `datalad run`